From 214bb8f6db52e4a79454fafdf4865f57f2af4053 Mon Sep 17 00:00:00 2001
From: Daniel Fiala <daniel@openssl.org>
Date: Tue, 27 Sep 2022 06:01:25 +0200
Subject: [PATCH 201/209] Disable printf format checking on MinGW

Fixes openssl#19185

Reviewed-by: Richard Levitte <levitte@openssl.org>
Reviewed-by: Tomas Mraz <tomas@openssl.org>
Reviewed-by: Paul Dale <pauli@openssl.org>
Reviewed-by: Hugo Landau <hlandau@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/19292)

(cherry picked from commit a1de5eb88479515535e5de090ded800455c3d4a7)
---
 include/openssl/bio.h.in | 1 +
 test/testutil.h          | 4 +++-
 test/testutil/output.h   | 1 +
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/openssl/bio.h.in b/include/openssl/bio.h.in
index 651843b629..c521e41e4a 100644
--- a/include/openssl/bio.h.in
+++ b/include/openssl/bio.h.in
@@ -796,6 +796,7 @@ void BIO_copy_next_retry(BIO *b);
 
 # define ossl_bio__attr__(x)
 # if defined(__GNUC__) && defined(__STDC_VERSION__) \
+    && !defined(__MINGW32__) && !defined(__MINGW64__) \
     && !defined(__APPLE__)
     /*
      * Because we support the 'z' modifier, which made its appearance in C99,
diff --git a/test/testutil.h b/test/testutil.h
index 1d9afffb58..d621204557 100644
--- a/test/testutil.h
+++ b/test/testutil.h
@@ -285,7 +285,9 @@ const OPTIONS *test_get_options(void);
  */
 
 # define PRINTF_FORMAT(a, b)
-# if defined(__GNUC__) && defined(__STDC_VERSION__)
+# if defined(__GNUC__) && defined(__STDC_VERSION__) \
+    && !defined(__MINGW32__) && !defined(__MINGW64__) \
+    && !defined(__APPLE__)
   /*
    * Because we support the 'z' modifier, which made its appearance in C99,
    * we can't use __attribute__ with pre C99 dialects.
diff --git a/test/testutil/output.h b/test/testutil/output.h
index 5e4f4c2390..0e4de09704 100644
--- a/test/testutil/output.h
+++ b/test/testutil/output.h
@@ -14,6 +14,7 @@
 
 # define ossl_test__attr__(x)
 # if defined(__GNUC__) && defined(__STDC_VERSION__) \
+    && !defined(__MINGW32__) && !defined(__MINGW64__) \
     && !defined(__APPLE__)
     /*
      * Because we support the 'z' modifier, which made its appearance in C99,
-- 
2.34.1

